src/stores/WalletStore.ts

```diff
@@ -38,24 +37,95 @@ class WalletStore extends PersistableStore {
     return web3Modal.cachedProvider
   }
 
-  async connect(clearCachedProvider = false) {
+  private async getUserNetworkName() {
```

Comment:
Provider may be `undefined` (types say that it can't, but actually it's not initialized before `.connect` happens), if you try to call this method without knowing this moment, you'll fail. Let's pass the `provider` here as non-optional argument

Reply:
ty, fixed